### DISCUSSION TITLE
Limited number of user rules for websites not covered uBOL lists

### DISCUSSION THREAD

**Discussion by LennyFox** [uBlockOrigin/uBOL-home#60] [Ideas]
I understand that uBOL's is designed as a lite version of uBO.  Requesting for extra features goes against this design philosophy. 

The idea I would like you to consider is: 

Add an option for user specified block rules (similar to uBO's My Filter's)  with the following limitations
1. User may enter a maximum of say 50 or 100 rules 
2. Only simple static rules (e.g. ||website.org/*/javascript.js^$script or ||adserver.nl/preload-banners^$frame)
3. Only simple cosmetic rules (e.g. website.org##.ad_placement or website.org###banner)
4. User rules are always website specific
5. User rules are only applied when complete mode is selected (as default or for a specific website)

For websites not covered by the uBOL default lists, this small number of user writeable ABP format rules would be a valuable addition to uBOL. 

Complements to uBOL, I really like the slider (i am using basic as default and apply complete only for a dozen websites).

**Comment by iam-py-test** [comment]
> Add an option for user specified block rules (similar to uBO's My Filter's) with the following limitations

Sadly, this has been declined in that past: https://github.com/uBlockOrigin/uBOL-issues/issues/17

**Comment by LennyFox** [comment]
@iam-py-test and @SampeiNihira 

@Kees1958 proposed a protection slider in uBO, it was declined and look what we have got now in uBOL (a slider :-) 

[Quote of @gorhill]
To anyone who doesn't like it, remind them that if you don't say No now, there's a good chance there won't even be an opportunity for a Yes later, if you're burned out!
[/Quote]

One person with strange ideas is just a fool, two or more people sharing this same idea are the start of a community :-)

**Reply by gorhill** [reply]
> Kees1958 proposed a protection slider in uBO, it was declined and look what we have got now in uBOL (a slider :-)

I didn't have this proposal in mind when I implemented the user interface in uBOL, this came after experimenting a whole lot about how to best deal with permissions levels which are core to uBOL.

As for your opening comment, I prefer not to go in that direction, this would require to integrate the full static network filtering engine in uBOL for the sake of compiling filters into DNR rules, and additionally, this doesn't work for cosmetic/scriptlet filtering since these requires to register _files_ to inject.

**Reply by stephenhawk8054** [reply]
> One person with strange ideas is just a fool, two or more people sharing this same idea are the start of a community :-)

A community should help in various ways: coding, maintaining the filter lists, spreading knowledge... Just suggesting ideas and throwing the rest to the dev will just lead to that blog: burnout.

**Reply by Kees1958** [reply]
@gorhill 

What are your thoughts of allowing user rules in the format of DNR?


@stephenhawk8054 

I wish you strength to recover from your burn out
